crates/storage/provider/src/providers/database/mod.rs

```diff
@@ -82,10 +83,12 @@ impl<DB> ProviderFactory<DB> {
     /// database provider comes with a shared snapshot provider
     pub fn with_snapshots(
         mut self,
+        snapshots_path: PathBuf,
         highest_snapshot_tracker: watch::Receiver<Option<HighestSnapshots>>,
     ) -> Self {
         self.snapshot_provider = Some(Arc::new(
-            SnapshotProvider::default().with_highest_tracker(Some(highest_snapshot_tracker)),
+            SnapshotProvider::new(snapshots_path)
+                .with_highest_tracker(Some(highest_snapshot_tracker)),
```

Comment:
I think these arguments became too verbose, can we just pass a `Arc<SnapshotProvider>` as an argument?